aca_preview/roll_optimize.py

```diff
@@ -0,0 +1,328 @@
+# coding: utf-8
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""
+Roll optimization during preliminary review of ACA catalogs.
+"""
+from copy import deepcopy
+from pathlib import Path
+import numpy as np
+
+from astropy.table import Table, vstack
+from chandra_aca.star_probs import acq_success_prob
+from chandra_aca.transform import (radec_to_yagzag, yagzag_to_pixels,
+                                   calc_aca_from_targ, calc_targ_from_aca,
+                                   snr_mag_for_t_ccd)
+from Quaternion import Quat
+
+import proseco.characteristics as ACA
+from proseco import get_aca_catalog
+
+
+ROLL_TABLE = Table.read(str(Path(__file__).parent / 'pitch_rolldev.csv'), format='ascii.basic',
+                        guess=False, delimiter=',')
+
+
+# Grab from chandra_aca PR #71
+def guide_count(mags, t_ccd, is_ER=False):
+    """Calculate a guide star fractional count/metric using signal-to-noise scaled
+    mag thresholds.
+    This uses a modification of the guide star fractional counts that were
+    suggested at the 7-Mar-2018 SSAWG and agreed upon at the 21-Mar-2018
+    SSAWG.  The implementation here does a piecewise linear interpolation
+    between the reference mag - fractional count points instead of the
+    original "threshold interpolation" (nearest neighbor mag <= reference
+    mag).  Approved at 16-Jan-2019 SSAWG.
+    One feature is the slight incline in the guide_count curve from 1.0005 at
+    mag=6.0 to 1.0 at mag=10.0.  This does not show up in standard outputs
+    of guide_counts to two decimal places (8 * 0.0005 = 0.004), but helps with
+    minimization.
+    :returns: fractional count
+    """
+    # Generate interpolation curve for the specified input ``t_ccd``
+    ref_t_ccd = -10.9
+    ref_mags0 = np.array([10.0, 10.2, 10.3, 10.4]) - (1.2 if is_ER else 0.0)
```

Comment:
That use of is_ER does not look like it is from PR #71 and probably deserves something in the docstring if it is staying?

Reply:
To me it would make more sense to replace is_ER option with the value to be subtracted.

Reply:
This isn't really a public method so we can change it later if that makes sense.  

For now it keeps the overall code cleaner since it keeps the "how to count for ERs" logic in one place (here) instead of repeating it three times in the upstream places that call it (in aca_preview).